nn/cond.py

```diff
@@ -0,0 +1,155 @@
+"""
+Conditional logic
+
+https://github.com/rwth-i6/returnn_common/issues/24
+"""
+
+from tensorflow.python.util import nest
+from .. import nn
+
+
+class Cond:
+  """
+  Conditional branching. Basically behaves like ``if ... else ...``.
+  Only one branch will be executed, and the condition needs to be a bool scalar.
+  This wraps to :class:`CondLayer` in RETURNN and to ``tf.cond`` in TensorFlow.
+
+  Example::
+
+      with Cond(cond) as cond_obj:
+        y = mod_true_case(x)
+        cond_obj.else(y)
+        y = mod_false_case(x)
+        y = cond_obj.end(y)
```

Comment:
This is outdated, and not what this PR actually implements, right?

Reply:
Yes, I'm going to update this. Actually I was still not really sure which API to implement and just wanted to test it as part of the test cases. Some suggestion have been made in #24. Although maybe I stick to the current implementation now. I think it's fine.